modules/terraform/workspace.go

```diff
@@ -60,3 +60,49 @@ func nameMatchesWorkspace(name string, workspace string) bool {
 	match, _ := regexp.MatchString(fmt.Sprintf("^\\*?\\s*%s$", name), workspace)
 	return match
 }
+
+// WorkspaceDelete removes the specified terraform workspace with the given options.
+// It returns the name of the current workspace AFTER deletion, and the returned error (that can be nil).
+// If the workspace to delete is the current one, then it tries to switch to the "default" workspace.
+// Deleting the workspace "default" is not supported.
+func WorkspaceDeleteE(t testing.TestingT, options *Options, name string) (string, error) {
+	currentWorkspace := RunTerraformCommand(t, options, "workspace", "show")
+
+	if name == "default" {
+		return currentWorkspace, fmt.Errorf("Deleting the workspace 'default' is not supported")
+	}
+
+	out, err := RunTerraformCommandE(t, options, "workspace", "list")
+	if err != nil {
+		return currentWorkspace, err
+	}
+	if !isExistingWorkspace(out, name) {
+		return currentWorkspace, fmt.Errorf("The workspace %q does not exist.", name)
+	}
+
+	// Switch workspace before deleting if it is the current
+	if currentWorkspace == name {
+		currentWorkspace = WorkspaceSelectOrNew(t, options, "default")
+	}
+
+	// delete workspace
+	_, err = RunTerraformCommandE(t, options, "workspace", "delete", name)
+
+	return currentWorkspace, err
+}
+
+// WorkspaceDelete removes the specified terraform workspace with the given options.
+// It returns the name of the current workspace AFTER deletion.
+// If the workspace to delete is the current one, then it tries to switch to the "default" workspace.
+// Deleting the workspace "default" is not supported and only return an empty string (to avoid a fatal error).
+func WorkspaceDelete(t testing.TestingT, options *Options, name string) string {
+	if name == "default" {
+		return name
+	}
```

Comment:
Why does this method differ from the `E` version with regards to the default workspace?

Reply:
I'm not sure about the expected behavior of this method as it does not output. Could I have your advise?

If consumer calls `terraform.WorkspaceDelete(t, opts, "default")`:

* Should it return the string `default`, and the consumer should take care of comparing it to the input parameter? Or should it return an empty string to mark the "edge case"?

Reply:
Shouldn't it just call `WorkspaceDeleteE` directly and return/use whatever that function returns?

Reply:
Looks like this never got resolved?

Reply:
I'm sorry, I missed the comment with the folded reviews. Let me check and fix (and I'll check all the others feedbacks to ensure I did not miss another). Thanks for your eagle eyes!

Reply:
You are correct: this code should returns whatever `WorkspaceDeleteE` returns. I don't know why I over-complicated it 😅